### PR TITLE
Add a display_data way of displaying widgets

### PR DESCRIPTION
**Pull request by jasongrout**
This is a step on the roadmap for widgets in JupyterLab, as discussed on https://github.com/jupyter/jupyter-js-notebook/issues/34. 

CC @jdfreder, @SylvainCorlay, @afshin

### CODE REVIEW

**Comment by jasongrout** [comment]
Note that this is a work in progress, since it will look ugly in the current notebook. I can make the display *not* have a text/plain type, so it would be invisible in the current notebook, so we can have both methods for a while. What do you think?

File: ipywidgets/widgets/widget.py [hunk 1]
```diff
@@ -472,7 +473,21 @@ def loud_error(message):
...
+            # below works
             self._send({"method": "display"})
+            
+            # The vnd.jupyter.widget mimetype has not been registered yet. See the 
+            # registration process and naming convention at http://tools.ietf.org/html/rfc6838 
+            # and the currently registered mimetypes at 
+            # http://www.iana.org/assignments/media-types/media-types.xhtml
+            data = {
+                'text/plain': 'A Jupyter widget',
+                'application/vnd.jupyter.widget': self._model_id
```
Comment (jdfreder) [inline ipywidgets/widgets/widget.py]: vnd?

Comment (afshin) [inline ipywidgets/widgets/widget.py]: > Vendor tree
> Vendor tree is used for media types associated with publicly available products. It uses "vnd." facet.

via https://en.wikipedia.org/wiki/Media_type

Comment (jdfreder) [inline ipywidgets/widgets/widget.py]: Thanks

**Comment by jasongrout** [comment]
@SylvainCorlay, @jdfreder - this should be enough for me to work on displaying widgets in the output areas. It's fine to leave this not merged until I have a complete working example...

**Comment by jasongrout** [comment]
@jdfreder, @SylvainCorlay - should I target a different branch now?

**Comment by SylvainCorlay** [comment]
@jasongrout I think that you should target 6.x.

**Comment by jasongrout** [comment]
Closing this to open a new PR targeted at the 6.x branch.

**Pull request closed without merging**